### ISSUE TITLE
MariaDB primary key 

### ISSUE THREAD

**Issue by shiizeen** [ansible-semaphore/semaphore#1824]
Hello there ! I have an issue when trying to setuping Semaphore Ansible with MariaDB (Galera Cluster) 

Here's the output of my error:  

``` time="2024-03-11T16:07:57Z" level=warning msg="\n ERR! Query: CREATE TABLE `event` (\n  `project_id` int DEFAULT NULL,\n  `object_id` int DEFAULT NULL,\n  `object_type` varchar(20) DEFAULT '',\n  `description` text,\n  `created` datetime NOT NULL\n)\n\n"
time="2024-03-11T16:07:57Z" level=fatal msg="Error 1173: This table type requires a primary key"```

And here's another one, when this time, i'm trying to create the table by myself : 

```2024-03-11T16:17:00.180329587Z time="2024-03-11T16:17:00Z" level=warning msg="\n ERR! Query: CREATE TABLE `event` (\n  `project_id` int DEFAULT NULL,\n  `object_id` int DEFAULT NULL,\n  `object_type` varchar(20) DEFAULT '',\n  `description` text,\n  `created` datetime NOT NULL\n)\n\n"
2024-03-11T16:17:00.180369427Z time="2024-03-11T16:17:00Z" level=fatal msg="Error 1050: Table 'event' already exists"``` 

MariaDB version : 10.5.23-MariaDB 
OS : Debian 11

Thank's for your help :)

**Comment by fiftin** [comment]
I use MariaDB 10.10.2 and I don't have this issue...

It is very old migration.

**Comment by tboerger** [comment]
@fiftinI just a guess, it's related to galera clustering which requires primary keys.

**Comment by shiizeen** [comment]
@tboerger , I guess too. I try on a standalone MariaDB and it's working properly. 

It can be a good thing to add some Galera compatibility ? :) 

Thx for your help !

**Comment by tboerger** [comment]
IMHO it should be possible to add a migration, but the problem is the initial migration based on galera. Maybe we need to alter the old migration and add a new migration which got some conditional query...

**Comment by fiftin** [comment]
@tboerger yes, I think about this long time :) We need refactor migrations.
